modules/mono/editor/bindings_generator.cpp

```diff
@@ -3919,57 +3919,67 @@ void BindingsGenerator::_initialize() {
 }
 
 static String generate_all_glue_option = "--generate-mono-glue";
-
-static void handle_cmdline_options(String glue_dir_path) {
-	BindingsGenerator bindings_generator;
-	bindings_generator.set_log_print_enabled(true);
-
-	if (!bindings_generator.is_initialized()) {
-		ERR_PRINT("Failed to initialize the bindings generator");
-		return;
-	}
-
-	CRASH_COND(glue_dir_path.is_empty());
-
-	if (bindings_generator.generate_cs_api(glue_dir_path.path_join(API_SOLUTION_NAME)) != OK) {
-		ERR_PRINT(generate_all_glue_option + ": Failed to generate the C# API.");
-	}
-}
+bool BindingsGenerator::generating_glue;
+String BindingsGenerator::glue_dir_path;
 
 static void cleanup_and_exit_godot() {
 	// Exit once done
 	Main::cleanup(true);
 	::exit(0);
 }
 
-void BindingsGenerator::handle_cmdline_args(const List<String> &p_cmdline_args) {
-	String glue_dir_path;
-
+void BindingsGenerator::process_cmdline(const List<String> &p_cmdline_args) {
 	const List<String>::Element *elem = p_cmdline_args.front();
 
 	while (elem) {
 		if (elem->get() == generate_all_glue_option) {
-			const List<String>::Element *path_elem = elem->next();
+			elem = elem->next();
 
-			if (path_elem) {
-				glue_dir_path = path_elem->get();
-				elem = elem->next();
-			} else {
+			if (elem) {
+				glue_dir_path = elem->get();
+			}
+
+			if (glue_dir_path.is_empty()) {
 				ERR_PRINT(generate_all_glue_option + ": No output directory specified (expected path to '{GODOT_ROOT}/modules/mono/glue').");
 				// Exit once done with invalid command line arguments
 				cleanup_and_exit_godot();
 			}
 
+			// valid glue path is set, enable glue generation
+			generating_glue = true;
+
 			break;
 		}
 
 		elem = elem->next();
 	}
+}
+
+// static trampoline to make sure the BindingsGenerator instance goes fully out of scope before exiting
+static void _generate(const String &p_glue_dir_path) {
+	BindingsGenerator bindings_generator;
+	bindings_generator.generate(p_glue_dir_path);
+}
+
+void BindingsGenerator::generate_bindings() {
+	_generate(glue_dir_path);
+
+	// Exit once done
+	cleanup_and_exit_godot();
+}
 
-	if (glue_dir_path.length()) {
-		handle_cmdline_options(glue_dir_path);
-		// Exit once done
-		cleanup_and_exit_godot();
+void BindingsGenerator::generate(const String &p_glue_dir_path) {
```

Comment:
This method seems to be the same as `handle_cmdline_options` so why rename it?

Reply:
Because this doesn't handle command line options anymore, it just generates bindings, and I thought it would be confusing to be called handle_cmdline_options when it's not doing that at all.

Reply:
I don't really see the difference between this method and the old `handle_cmdline_options` and we could argue that `p_glue_dir_path` is the option that's being handled which comes from a cmdline argument.